src/main/java/seedu/address/ui/ClientInfoPage.java

```diff
@@ -31,23 +64,203 @@
+        if (currentPage == null) {
+            currentPage = new ClientInfoPage();
+        }
+        return currentPage;
+    }
+
+    /**
+     * Updates the GUI with the lastest information about the client
+     */
+    public void update(Logic logic) {
+        if (this.client == null) {
+            return;
+        }
+        Optional<Client> optionalClient = logic.getAddressBook().getClientList().stream()
+                .filter(x->x.isUnique(this.client)).findFirst();
+        if (optionalClient.isPresent()) {
+            this.client = optionalClient.get();
+            initializeProfile();
+            initializeSchedule(logic.getAssociatedScheduleList(this.client));
+            initializeChart(logic.getPreferredWeightUnit(), logic.getAssociatedScheduleList(this.client));
+        }
+    }
+
+    public static ClientInfoPage getClientInfoPage(Client client, List<Schedule> pastSchedules, WeightUnit weightUnit) {
+        return new ClientInfoPage(client, pastSchedules, weightUnit);
+    }
+
+    private void initializeProfile() {
         this.lblName.setText(client.getName().fullName);
         this.lblPhone.setText(client.getPhone().value);
         this.lblEmail.setText(client.getEmail().value);
         this.lblAddress.setText(client.getAddress().value);
+        tags.getChildren().clear();
         client.getTags().stream()
                 .sorted(Comparator.comparing(tag -> tag.tagName))
                 .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
+        try {
+            this.imgProfile.setImage(retrieveImage());
+        } catch (NullPointerException | IllegalArgumentException e) {
+            logger.info("Invalid image url, using default image\nException: " + e);
+        }
+
+    }
+
+    private Image retrieveImage() {
+        // Set image based on client's name first character. Skipping if invalid url found.
+        // Just to make the app a bit nicer with real human image
+        return new Image("/images/profile-"
+                + ((client.getName().fullName.toLowerCase().charAt(0) - 'a') / 6 + 1) + ".jpg");
+    }
+
+    @SuppressWarnings("unchecked")
+    private void initializeChart(WeightUnit weightUnit, List<Schedule> pastSchedules) {
+        //x is date (at the bottom)
+        //y is weight (at the left)
+        XYChart.Series<String, Number> xy = new XYChart.Series<>();
+
+        List<XYChart.Data<String, Number>> xyList = pastSchedules.stream()
+                .filter(x -> x.getClient().equals(client) && !x.getWeight().equals(Weight.getDefaultWeight()))
+                .sorted() //use default comparator
+                .map(x -> {
+                    XYChart.Data<String, Number> data = new XYChart.Data<>(
+                            SessionParserUtil.parseDateTimeToString(x.getSession().getStartTime()),
+                            weightUnit.isPoundUnit()
+                                    ? getKgInPound(x.getWeight().getWeight())
```

Comment:
this is still understandable to me, but would it be confusing to call 2 methods with the same name? For other objects, usually we call .getName().value (but value shdn't be public).
so another possible naming is getValue().

Reply:
i'll let @tanweijie123 decide this one